index.rst

```diff
@@ -240,6 +240,101 @@ fitSrcMatchAstrometry
+- Task labels and dataset type names are for "slots", not specific tasks or connections - usually those are 1-1, but when they are not, the label and dataset type names should remain fixed when a different task is swapped in (our "solveAstrometry" task label slot could be satisfied by either jointcal or GBDES).
+  Whether the same is true of the outputs is an open question we'd like to discuss.
+- "initial" catalogs are not (ever) SDM-standardized, while final catalogs are always SDM-standardized before they are written out.
+  This may present a challenge for analysis tools that want to be able to run on either, but I'm hoping we can minimize analysis of the initial things, as we don't care how good they are in the end.
+- Do include a "final" prefix on dataset types that represent the best version of things that are nevertheless temporaries that will not be retained.
+  Do not add any prefix to dataset types that will be retained for public access.
+- Convert source and object catalogs to Parquet before ever persisting them; only use SourceCatalog/FITS to hold Footprints.
+  We will retain the ``slot_*`` column names and not the underlying names they point to - downstream code should only be referring to the slot columns anyway, and in many cases we won't run multiple algorithms that could satisfy the slot.
+  In addition, note that the (final) ``source`` catalogs will be SDM-standardized before being written, so the the ``slot_*`` vs. underlying name question is moot there.
+
+Task and dataset type notes
+"""""""""""""""""""""""""""
+
+bootstrapImage
+   Initial background subtraction and detection of bright stars (galaxies are considered a nuisance here).
+   Initial versions of everything - astrometry, photometry, probably PSFs.
+   Probably aperture corrections of some kind, but targeted specifically at making compensated apertures work for FGCM.
+   These are all attached to its image output, ``initial_pvi``, which is a lot like today's ``calexp``.
+   This will be a fluence image with nJy pixel units, but how much of that is done by this task vs. ISR is TBD.
+   Its output catalog, ``initial_source_detector``, will be converted to Parquet before it is written, but not SDM-standardized.
+   Footprints will be written to a separate ``SourceCatalog`` dataset, ``initial_footprints``.
+   Whether this task does multiple detection rounds (to iterate on CR detection, the source detection filter, or the detection threshold) is TBD; we'd like to minimize that.
+   We would like to avoid running a deblender here (if we do, it will have to happen after we have the PSF model), but this depends on how this performs on crowded fields.
+   We are not thrilled with the name of this task and would love ideas for improvements (which should be coordinated with the names for ``bootstrapVisit``, ``finalizeImage``, ``compressImage``, and ``rebuildImage``).
```

Comment:
I think this misses the legitimate use case of calexp now, that of a "good enough" image for things, even if it is not the final calibration (especially for pipelines that do not run FGCM) but I may not be following how that use case plays out in the proposed graph

Reply:
In pipelines that do not produce a better version of the image produced, the image produced here should have a name that does not imply that it will be superseded.  But I want to focus on the actual DRP pipeline here.